src/java/org/apache/cassandra/serializers/UTF8Serializer.java

```diff
@@ -57,8 +60,51 @@ static <V> boolean validate(V value, ValueAccessor<V> accessor)
             if (value == null)
                 return false;
 
-            int b = 0;
-            int offset = 0;
+            // perf optimizations:
+            // 1) avoid bymorhic/megamorphic calls via ValueAccessor
+            // 2) use a simplified logic to handle ASCII prefixed String scenario faster
+            if (accessor == ByteArrayAccessor.instance)
+            {
+                byte[] valueAsArray = accessor.toArray(value);
+                for (int i = 0; i < valueAsArray.length; i++)
+                {
+                    if (valueAsArray[i] < 0)
+                        return validateSlowPath(value, accessor, i);
+                }
+                return true;
```

Comment:
I see this block of code is repeated across this method, can you please extract it to a helper to reduce code duplication and for better readability?